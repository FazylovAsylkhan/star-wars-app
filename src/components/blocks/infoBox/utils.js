function getArrPlanetsData(objectData) {
  const arrInfoHomeworld = [
    `Population: ${objectData?.population}`,
    `Diameter: ${objectData?.diameter}`,
    `Rotation period: ${objectData?.rotation_period}`,
    `Climate: ${objectData?.climate}`,
    `Gravity: ${objectData?.gravity}`,
    `Terrain: ${objectData?.terrain}`,
    `Surface water: ${objectData?.surface_water}`,
    `Orbital period: ${objectData?.orbital_period}`,
  ];
  return arrInfoHomeworld;
}
function getArrStarshipsData(objectData) {
  const arrInfoHomeworld = [
    `model: ${objectData?.model}`,
    `MGLT: ${objectData?.MGLT}`,
    `Consumables: ${objectData?.consumables}`,
    `Crew: ${objectData?.crew}`,
    `Hyperdrive rating: ${objectData?.hyperdrive_rating}`,
    `Length: ${objectData?.hyperdrive_rating}`,
    `Manufacturer: ${objectData?.manufacturer}`,
    `Max atmosphering speed: ${objectData?.max_atmosphering_speed}`,
    `Passengers: ${objectData?.passengers}`,
    `Starship class: ${objectData?.starship_class}`,
  ];
  return arrInfoHomeworld;
}
function getArrPeopleData(objectData) {
  const arrInfoPerson = [
    `Gender: ${objectData?.gender}`,
    `Height: ${objectData?.height}`,
    `Mass: ${objectData?.mass}`,
    `Hair color: ${objectData?.hair_color}`,
    `Skin color: ${objectData?.skin_color}`,
    `Eye color: ${objectData?.eye_color}`,
    `Birth year: ${objectData?.birth_year}`,
  ];
  return arrInfoPerson;
}

export function getArrHomeworldData(objectData) {
  const arrInfoHomeworld = [
    `Homeworld: ${objectData?.name}`,
    `Population: ${objectData?.population}`,
    `Diameter: ${objectData?.diameter}`,
    `Rotation period: ${objectData?.rotation_period}`,
  ];
  return arrInfoHomeworld;
}

export function getRelativeUrl(url) {
  return url.split("/").splice(4);
}
export function getHandlerDataFor(namePage) {
  switch (namePage) {
    case "people":
      return getArrPeopleData;
    case "planets":
      return getArrPlanetsData;
    case "starships":
      return getArrStarshipsData;
    default:
      return getArrHomeworldData;
  }
}
