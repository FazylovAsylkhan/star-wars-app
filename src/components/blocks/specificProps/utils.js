function getAllExistSpecificProps(data) {
  return [
    { name: "Starships", values: data.starships },
    { name: "Vehicles", values: data.vehicles },
    { name: "Films", values: data.films },
    { name: "Species", values: data.species },
    { name: "Residents", values: data.residents },
    { name: "Pilots", values: data.pilots },
  ];
}

function filterProps(props) {
  const filteredProps = props.filter((prop) => {
    const { values } = prop;
    const IsExitValues = values !== undefined;
    const IsNotEmptyValues = values?.length > 0;

    return IsExitValues && IsNotEmptyValues ? prop : null;
  });

  return filteredProps;
}

const getFilteredProps = (data) => {
  let result = null;
  if (data) {
    const allExistSpecificProps = getAllExistSpecificProps(data);
    result = filterProps(allExistSpecificProps);
  }
  return result;
};

export default getFilteredProps;
