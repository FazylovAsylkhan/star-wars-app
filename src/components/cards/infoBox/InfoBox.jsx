/* eslint-disable prefer-destructuring */
import React from "react";
import s from "./infoBox.module.scss";

function InfoBox({ results }) {
  const list = results.filter((item, index) => index < 8);

  return (
    list.length > 0 && (
      <ul className={s.list}>
        {list.map(([props, value]) => (
          <li key={props}>
            <b>{props}: </b>
            {value}
          </li>
        ))}
      </ul>
    )
  );
}

export default InfoBox;
