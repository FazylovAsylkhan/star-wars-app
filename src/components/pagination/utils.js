import s from "../button/button.module.scss";

function addActiveClass(url = null) {
  const currentNumberOfPage = url ? url.split("=")[1] : "1";
  const pagination = document.getElementById("pagination");
  const btns = Array.from(pagination.getElementsByTagName("button"));
  btns.forEach((btn) => {
    btn.classList.remove(s.activeBtn);
    if (btn.textContent === currentNumberOfPage) btn.classList.add(s.activeBtn);
  });
}

export default addActiveClass;
