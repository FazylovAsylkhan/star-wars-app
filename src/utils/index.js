export const transformDataToArray = (item) => Object.entries(item);
export const transformSubDataToArray = (obj) =>
  obj.results.map((item) => transformDataToArray(item));
