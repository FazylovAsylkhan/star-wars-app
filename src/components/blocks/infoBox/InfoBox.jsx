import React from "react";
import Button from "../../button/Button";
import List from "../../list";
import { getHandlerDataFor, getRelativeUrl } from "./utils";
import s from "./infoBox.module.scss";

function InfoBox({ url, handlerClick = null }) {
  const [namePage] = getRelativeUrl(url);
  const handler = handlerClick
    ? getHandlerDataFor(namePage)
    : getHandlerDataFor();

  return (
    <div className={s.infoBox}>
      <List url={url} callback={handler} />
      {handlerClick && (
        <Button type="error" callback={() => handlerClick()}>
          Throw Error
        </Button>
      )}
    </div>
  );
}

export default InfoBox;
