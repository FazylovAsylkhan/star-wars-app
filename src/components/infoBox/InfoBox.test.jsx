/* eslint-disable no-undef */
import React from "react";
import { screen } from "@testing-library/react";
import { renderWithRouter } from "../../tests/testUtils";
import { transformDataToArray } from "../../utils";
import testData from "../../tests/testData";
import InfoBox from "./InfoBox";

describe("InfoBox component works", () => {
  it("should render InfoBox component", async () => {
    const handledData = transformDataToArray(testData.results[0]);
    renderWithRouter(<InfoBox results={handledData} />);
    const infoBox = screen.getByText(/Luke Skywalker 1/i);
    expect(infoBox).toBeInTheDocument();
    expect(infoBox).toMatchSnapshot();
  });
});
