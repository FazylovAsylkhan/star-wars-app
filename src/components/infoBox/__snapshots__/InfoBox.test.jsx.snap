// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox component works should render InfoBox component 1`] = `
<li>
  <b>
    name
    : 
  </b>
  Luke Skywalker 1
</li>
`;
